zzk/service/servicestate.go

```diff
@@ -127,7 +127,7 @@ func (l *ServiceListener) Spawn(shutdown <-chan interface{}, serviceID string) {
 			}
 
 			// Synchronize the number of service states
-			if _, ok := l.Sync(sDat.Locked, sDat.Service, reqs); !ok {
+			if _, ok := l.Sync(sDat.Locked, sDat.AsService(), reqs); !ok {
```

Comment:
Changed it to pass sdat, and modified subsequent calls to handle a ServiceNode. in  scheduler/leader.go, added a call to retrieve the full service from elastic. Per Glenn (and me), this change definitely needs a review from you for correctness.